xc/xenops_server_xen.ml

```diff
@@ -978,7 +995,7 @@ module VM = struct
         let _ = DB.update k (function
             | Some x ->
               debug "VM = %s; reloading stored domain-level configuration" vm.Vm.id;
-              Some x
+              Some VmExtra.{persistent = (DB.revision_of k x.VmExtra.persistent)}
```

Comment:
`create_exn` seems like the wrong place to be doing this upgrade. Why is this necessary _as well_ as the `set_internal_state` change?

Reply:
I think the the change in create_exn was a convenient hook to update the xenopsd db whenever we do receive_memory and resume. I later patched set_internal_state so that the persistent state is using the latest upgraded version as early as possible. In tests, either one of them is sufficient. The profile upgrade is idempotent so calling more than once is not a problem. However, it looks like it may be better to patch only set_internal_state, because in this case the change will also be compatible with upgrades that are not idempotent.